versioned/version0_3_2.ts

```diff
@@ -0,0 +1,400 @@
+    );
+
+    if (transacValue === -1) {
+      return [true, op.opHash];
+    }
+
+    const confirmationValue = await promiseWithTimeout(
+      op.confirmation(1),
+      DEFAULT_TIMEOUT
+    );
+
+    if (confirmationValue === -1) {
+      return [true, op.opHash];
+    }
+
+    return [false, op.opHash];
+  }
+  async signProposal(
+    cc: WalletContract,
+    t: TezosToolkit,
+    proposal: number,
+    result: boolean | undefined,
+    resolve: boolean
+  ): Promise<timeoutAndHash> {
+    const proposals: { proposals: BigMapAbstraction } = await cc.storage();
+    const proposalId = num2PaddedHex(proposal);
+    const prop: any = await proposals.proposals.get(proposalId);
+    const batch = t.wallet.batch();
+
+    const proposalData = arrayProposalSchema.Encode(prop.contents);
+
+    const proposalBytes = packDataBytes(proposalData, proposalsType).bytes;
+
+    if (typeof result != "undefined") {
+      batch.withContractCall(
+        cc.methodsObject.sign_proposal({
+          agreement: result,
+          challenge_id: proposalId,
+          payload: proposalBytes,
+        })
+      );
+    }
+    if (resolve) {
+      batch.withContractCall(
+        // resolve proposal
+        cc.methods.proof_of_event_challenge(proposalId, proposalBytes)
+      );
+    }
+    let op = await batch.send();
+
+    const confirmationValue = await promiseWithTimeout(
+      op.confirmation(1),
+      DEFAULT_TIMEOUT
+    );
+
+    return [confirmationValue === -1, op.opHash];
+  }
+
+  async submitSettingsProposals(
+    cc: Contract,
+    t: TezosToolkit,
+    ops: ownersForm[]
+  ): Promise<timeoutAndHash> {
+    let content = ops
+      .map(v => {
+        if ("addOwners" in v) {
+          return { add_owners: v.addOwners };
+        } else if ("removeOwners" in v) {
+          return { remove_owners: v.removeOwners };
+        } else if ("changeThreshold" in v) {
+          return { adjust_threshold: Number(v.changeThreshold) };
+        } else if ("adjustEffectivePeriod" in v) {
+          return { adjust_effective_period: v.adjustEffectivePeriod };
+        } else {
+          return v;
+        }
+      })
+      .filter(x => !!x);
+
+    let params = cc.methods.create_proposal(content).toTransferParams();
+
+    let op = await t.wallet.transfer(params).send();
+
+    const transacValue = await promiseWithTimeout(
+      op.transactionOperation(),
+      DEFAULT_TIMEOUT
+    );
+
+    return [transacValue === -1, op.opHash];
+  }
+  static override toContractState(
+    contract: any,
+    balance: BigNumber
+  ): contractStorage {
+    let c: c1 = contract;
+    return {
+      balance: balance!.toString() || "0",
+      proposal_map: c.proposals.toString(),
+      proposal_counter: c.proposal_counter.toString(),
+      effective_period: c!.effective_period,
+      threshold: c!.threshold.toNumber()!,
+      owners: c!.owners!,
+      version: "0.3.2",
+    };
+  }
+  private static mapContent(content: content): proposalContent {
```

Comment:
`add_or_update_metadata` is not handled at all, even if we don't handle it on the UI side, I think it should be handled here just in case

Reply:
no, it will throw error. as we don't have any unit test here, it's no way to know if you do it right or wrong.  It will be better going with UI.

Reply:
So should we just ignore it for now?

Reply:
yes, for now, nothing will go wrong.